openmetadata-service/src/main/java/org/openmetadata/service/secrets/SecretsManagerFactory.java

```diff
@@ -43,18 +45,18 @@ public static SecretsManager createSecretsManager(SecretsManagerConfiguration co
         If for example we want to set the AWS SSM (non-managed) we configure
         the server as `secretsManager: aws-ssm` and set the Airflow env vars
         to connect to AWS SSM as specified in the docs:
-        https://docs.open-metadata.org/v1.0.0/deployment/secrets-manager/supported-implementations/aws-ssm-parameter-store
+        https://docs.open-metadata.org/deployment/secrets-manager/supported-implementations/aws-ssm-parameter-store
         */
-        secretsManager = NoopSecretsManager.getInstance(clusterName, secretsManagerProvider);
+        secretsManager = NoopSecretsManager.getInstance(secretsManagerProvider, secretsConfig);
```

Comment:
mmhh when trying it out it gets a bit ugly actually. We are not initializing all the secret managers just the one being requested by the config, and it would require some further changes on the parent class to allow getting the instances just right.

if it's ok, I'll keep the switch for now